### DISCUSSION TITLE
How e2e works in databag?

### DISCUSSION THREAD

**Discussion by RedNumber1** [balzack/databag#82] [Q&A]
So topics can be sealed but I can't add anyone. Also how do I seal one on one message?

**Comment by balzack** [comment]
To have a sealed topic (e2e) thread with someone, they need to have their sealing key (e2e) setup. If their key has not been setup, they won't show up in the member list when creating a topic.

There currently is the restriction that membership of a sealed topic cannot be changed once created. This was mostly done to minimize user error with sealed data. I can probably make this a configurable restriction if it seems too burdensome. 

Currently to have a 1-1 message be sealed, you have to create a topic and add that one person. I should probably make 'sealed' be the default 1-1 type when starting a DM topic from the contact list. I will add that to the backlog now.

Thanks for using the project! Please let me know if there are ways to make it more intuitive, or if there is missing functionality that you would like.

**Reply by RedNumber1** [reply]
OK thanks for the replay. I see my friend now he setup his key.
